### PR TITLE
Features: Completed Grid, Modal and Navigation components

### PR DESCRIPTION
**Pull request by koushikrp**
#Changes Made:
Grid component
 - Completed 2 col grid
 - Completed 3 col grid
 - Added docs for grid
Modal component
 - Completed simple modal component
 - Added docs for modal
Navigation component
- Completed simple navigation component
 - Added docs for navigation

### CODE REVIEW

**Review by Prabhav-Sharma — APPROVED** [review]

File: src/components/modal/modal.css [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+.modal-wrapper {
+	position: fixed;
+	height: 100%;
+	width: 100%;
+	left: 0;
+	top: 0;
+	z-index: 4;
+	background-color: rgba(116, 116, 116, 0.521);
```
Comment (pritam-kr) [inline src/components/modal/modal.css:30]: Use variable for this color .

File: src/components/modal/modal.css [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	display: flex;
+	justify-content: center;
+	align-items: center;
```
Comment (pritam-kr) [inline src/components/modal/modal.css:33]: you can utility classes for this but it's ok.

File: src/components/modal/modal.css [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	display: none;
+}
+
+.modal-content {
+	background-color: var(--color-light);
+	height: 70px;
```
Comment (pritam-kr) [inline src/components/modal/modal.css:39]: use relative units after 10px

**Comment by Prabhav-Sharma** [comment]
Try to raise one PR for one component. Clubbing them together in a single PR is not ideal.

File: src/components/modal/modal.html [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+		</main>
+		<script>
+			const modal = document.querySelector(".modal-wrapper");
+			const closeModal = () => {
+				modal.style.display = "none";
+			};
+			const showModal = () => {
+				modal.style.display = "flex";
+			};
+		</script>
```
Comment (pritam-kr) [inline src/components/modal/modal.html:154]: you can use a separate page for JS

**Review by pritam-kr — APPROVED** [review]
I would like to add some good thing
- you have written clean code
- naming is good
- code formatted is good

File: src/components/modal/modal.css (hunk 1 above)
Comment (aamrapaliwandhre) [inline src/components/modal/modal.css:30]: use color variable...

**Comment by koushikrp** [comment]
Thank You Pritam. Will make mentioned changes.

**Review by aamrapaliwandhre — APPROVED** [review]

File: src/components/modal/modal.css (hunk 1 above)
Comment (koushikrp) [reply src/components/modal/modal.css:30]: Okay Thanks bro

**Pull request merged**